docs/providers/aws/guide/functions.md

```diff
@@ -329,3 +329,28 @@ functions:
 The `onError` config currently only supports SNS topic arns due to a race condition when using SQS queue arns and updating the IAM role.
 
 We're working on a fix so that SQS queue arns are be supported in the future.
+
+## KMS Keys
+
+AWS Lambda uses [AWS Key Management Service (KMS)](https://aws.amazon.com/kms/) to encrypt your environment variables at rest.
+
+The function-level `awsKmsKeyArn` config variable enables you a way to define your own KMS key which should be used for encryption.
+
+```yml
+service: service-name
+provider:
+  name: aws
+  environment:
+    TABLE_NAME: tableName1
+
+functions:
+  hello:
+    handler: handler.hello
+    awsKmsKeyArn: arn:aws:kms:us-east-1:XXXXXX:key/some-hash
+    environment:
+      TABLE_NAME: tableName2
+```
+
+### Secrets using environment variables and KMS
+
+When storing secrets in environment variables, AWS [strongly suggests](http://docs.aws.amazon.com/lambda/latest/dg/env_variables.html#env-storing-sensitive-data) encrypting sensitive information. AWS provides a [tutorial](http://docs.aws.amazon.com/lambda/latest/dg/tutorial-env_console.html) on using KMS for this purpose.
```

Comment:
My plugin simplifies the process further.

- add `kmsKeyId: ${self:provider.stage}-my-service` under `custom` (I should probably rename it "kmsKeyAlias")
- add a set of custom properties to be `encrypted` 
- use the same property names in `environment` sections

Then, the plugin
- generates a new KMS Key if an existing key with matching alias does not exist
- sets a sensible policy on the KMS key when created
- takes unencrypted environment variables (protected by CI), encrypts using the KMS key and rewrites any matching environment variables in the functionObjects
- (attempts to) write the `kmsKeyArn` to the output template

https://github.com/nalbion/serverless-plugin-encrypted